utils/utils.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python3
+
+###########################################################################
+#
+#    Copyright 2024 Google LLC
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#            https://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+###########################################################################
+
+""" Utils Module for generic functions """
+
+import argparse
+import textwrap
+from configuration import Configuration
+
+
+def build_abcd_params_config(args: any) -> Configuration:
+    """Builds ABCD configuration with all the required parameters.
+
+    Args:
+        args: The parser arguments.
+    Returns:
+        config: The parameter configuration for ABCD.
+
+    """
+    config = Configuration()
+    config.set_parameters(
+        project_id=args.project_id,
+        project_zone=args.project_zone,
+        bucket_name=args.bucket_name,
+        knowledge_graph_api_key=args.knowledge_graph_api_key,
+        bigquery_dataset=args.bigquery_dataset,
+        bigquery_table=args.bigquery_table,
+        assessment_file=args.assessment_file,
+        use_annotations=args.use_annotations,
+        use_llms=args.use_llms,
+        verbose=args.verbose,
+    )
+    config.set_videos(args.video_uris)
+    config.set_brand_details(
+        brand_name=args.brand_name,
+        brand_variations=args.brand_variations,
+        products=args.branded_products,
+        products_categories=args.branded_products_categories,
+        call_to_actions=args.branded_call_to_actions,
+    )
+    return config
+
+
+def parse_args(arg_list: list[str] | None = None) -> None:
```

Comment:
No action for this one, but I'm partial to [click](https://click.palletsprojects.com/en/stable/) over argparse for testability and usability.